tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_all_opcodes_in_container.py

```diff
@@ -98,34 +100,285 @@ def sections(
             sections.append(
                 Section.Code(
                     code=Op.RETF,
-                    code_inputs=0,
                     code_outputs=0,
-                    max_stack_height=0,
                 )
             )
     sections.append(Section.Data("1122334455667788" * 4))
-    return sections
+
+    if opcode == Op.RETURNCONTRACT:
+        eof_code = Container(sections=sections, kind=ContainerKind.INITCODE)
+    else:
+        eof_code = Container(sections=sections)
+
+    eof_test(
+        data=eof_code,
+        expect_exception=(
+            None if opcode in valid_eof_opcodes else EOFException.UNDEFINED_INSTRUCTION
+        ),
+    )
 
 
 @pytest.mark.parametrize(
-    "opcode", [op for op in list(Op) + list(UndefinedOpcodes) if op != Op.RETF]
+    "opcode",
+    sorted(
+        valid_eof_opcodes
+        - halting_opcodes
+        - section_terminating_opcodes
+        - {Op.RJUMP, Op.RJUMPI, Op.RJUMPV}
+    ),
 )
-def test_all_opcodes_in_container(
+def test_all_invalid_terminating_opcodes(
     eof_test: EOFTestFiller,
-    sections: List[Section],
-    expect_exception: EOFException | None,
     opcode: Opcode,
 ):
     """
-    Test all opcodes inside valid container
-    257 because 0x5B is duplicated
+    Test all opcodes that are invalid as the last opcode in a container
     """
-    if opcode == Op.RETURNCONTRACT:
-        eof_code = Container(sections=sections, kind=ContainerKind.INITCODE)
+    if opcode.has_data_portion():
+        # Add the appropriate data portion to the opcode by using the get_item method.
+        # On the CALLF opcode we need to reference the second code section, hence the [1] index.
+        opcode = opcode[0] if opcode != Op.CALLF else opcode[1]
+
+    bytecode = (Op.PUSH0 * opcode.min_stack_height) + opcode
```

Comment:
This looks much cleaner than `opcode with data portion`

But will it conflict with line 146 ?

Reply:
I thought about this conflict for a while because after appending the data portion with the square brackets, if we compare it against the opcode without the data portion, they are not equal, so `Op.OPCODE[0x123] != Op.OPCODE`, which is a bit annoying.

We could do some fancy method like `Op.OPCODE.is_descendant(Op.OPCODE[0x123])` or similar, which could return true if the opcode is the same but with the data portion added, but I'm not convinced it's super necessary.

So that's why on 146 we compare against `Op.CALLF[1]` and not against `Op.CALLF`.